integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/standardpartition/StandardPartitionTableLoadingTestCase.scala

```diff
@@ -566,6 +567,18 @@ class StandardPartitionTableLoadingTestCase extends QueryTest with BeforeAndAfte
     assert(ex.getMessage().equalsIgnoreCase("Cannot use all columns for partition columns;"))
   }
 
+  test("test partition without merge index files for segment") {
+    sql("DROP TABLE IF EXISTS new_par")
+    CarbonProperties.getInstance()
+      .addProperty(CarbonCommonConstants.CARBON_MERGE_INDEX_IN_SEGMENT, "false")
+    sql("CREATE TABLE new_par (a INT, b INT) PARTITIONED BY (country STRING) STORED AS carbondata")
+    sql("INSERT INTO new_par PARTITION(country='India') SELECT 1,2")
+    sql("INSERT INTO new_par PARTITION(country='India') SELECT 3,4")
+    sql("INSERT INTO new_par PARTITION(country='China') SELECT 5,6")
+    sql("INSERT INTO new_par PARTITION(country='China') SELECT 7,8")
+    checkAnswer(sql("SELECT COUNT(*) FROM new_par"), Seq(Row(4)))
```

Comment:
please check for index files also

Reply:
I think, it is redundant to do that. If segment files are not present or index files are not present, this query check will fail.